charts/hdfs-namenode-k8s/templates/namenode-statefulset.yaml

```diff
@@ -94,8 +163,44 @@ spec:
               mountPath: /etc/security/
               readOnly: true
             {{- end }}
-      {{- if .Values.kerberosEnabled }}
       initContainers:
+        # Write the namenode bootstrap script which will start namenode daemons
+        # after conducting optional metadata initialization steps. The metadata
+        # initialization steps will take place in case the metadata dir is empty,
+        # which will be the case only for the very first run. The specific steps
+        # will differ depending on whether the namenode is active or standby.
+        # We also assume, for the very first run, namenode-0 will be active and
+        # namenode-1 will be standby as StatefulSet will launch namenode-0 first
+        # and zookeeper will determine the sole namenode to be the active one.
+        # For active namenode, the initialization steps will format the metadata,
+        # zookeeper dir and journal node data entries.
+        # For standby namenode, the initialization steps will simply receieve
+        # the first batch of metadata updates from the journal node.
+        # TODO: Support upgrade of metadata in case a new Hadoop version requires it.
+        - name: write-start-script
+          image: busybox:1.27.1
+          command: ['sh', '-c']
+          args:
+            - >-
+              (echo '#!/bin/bash -x' > $OUT);
+              (echo 'if [[ ! -d /hadoop/dfs/name/current && "$MY_POD" = hdfs-namenode-0 ]]' >> $OUT);
+              (echo 'then' >> $OUT);
+              (echo ' $HADOOP_PREFIX/bin/hdfs --config $HADOOP_CONF_DIR namenode -format -force -nonInteractive hdfs-k8s || exit 1' >> $OUT);
```

Comment:
I worry about this `-force` options doing something stupid, like wiping out data when it should not. Let me check if we can live without them.